imagebuildah/executor.go

```diff
@@ -289,18 +290,15 @@ func (b *Executor) resolveNameToImageRef(output string) (types.ImageReference, e
 	return imageRef, nil
 }
 
+// waitForStage waits for an entry to be added to terminatedStage indicating
+// that the specified stage has finished.  If there is no stage by that name,
+// then it will wait forever, so the caller should ensure that there really is
+// a stage with the specified name before calling it.
 func (b *Executor) waitForStage(ctx context.Context, name string) error {
-	stage := b.stages[name]
-	if stage == nil {
-		return errors.Errorf("unknown stage %q", name)
-	}
```

Comment:
Do we really want to remove this check?  With the code as is, I know it should always pass, but I think it's a good fail-safe for future calls no?

I'm always very leery about potential never-ending loops like this.  I've been trying to think of an escape hatch, but can't at the moment that makes sense.

Reply:
Okay, moved some things around.

Reply:
Thx!